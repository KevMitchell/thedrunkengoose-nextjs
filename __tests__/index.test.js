import { render, fireEvent } from '@testing-library/react'
import Index from '../pages/index'

jest.mock('../public/data/drinks.json', () => [
  {
    name: 'drink1',
    image: 'drink1.jpg',
    difficulty: 'Easy',
    base: 'Whiskey',
    flavour: ['Strong', 'Ginger'],
    ingredients: [
      { name: 'Whiskey (bourbon)', quantity: 50 },
      { name: 'Lemon juice', quantity: 15 }
    ]
  },
  {
    name: 'drink2',
    image: 'drink2.jpg',
    difficulty: 'Hard',
    base: 'Gin',
    flavour: ['Lemonade', 'Ginger'],
    virgin: true,
    ingredients: [
      { name: 'Lemonade', quantity: 50 }
    ]
  }
])

jest.mock('../public/data/instructions.json', () => ({
  drink1: ['inst1', 'inst2'],
  drink2: ['inst3', 'inst4']
}))

const renderComponent = () => {
  const { container, queryByText, getByTestId } = render(<Index />)

  return {
    container,
    queryByText,
    typeFilter: getByTestId('type-filter'),
    difficultyFilter: getByTestId('difficulty-filter'),
    baseFilter: getByTestId('base-filter'),
    flavourFilter: getByTestId('flavour-filter'),
    searchFilter: getByTestId('search')
  }
}

test('renders recipe section', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})

test('filters correctly on type', () => {
  const { queryByText, typeFilter } = renderComponent()

  fireEvent.change(typeFilter, { target: { value: 'alcoholic' } })
  expect(queryByText(/drink1/)).toBeInTheDocument()
  expect(queryByText(/drink2/)).not.toBeInTheDocument()

  fireEvent.change(typeFilter, { target: { value: 'virgin' } })
  expect(queryByText(/drink1/)).not.toBeInTheDocument()
  expect(queryByText(/drink2/)).toBeInTheDocument()
})

test('filters correctly on difficulty', () => {
  const { queryByText, difficultyFilter } = renderComponent()

  fireEvent.change(difficultyFilter, { target: { value: 'Easy' } })

  expect(queryByText(/drink1/)).toBeInTheDocument()
  expect(queryByText(/drink2/)).not.toBeInTheDocument()

  fireEvent.change(difficultyFilter, { target: { value: 'Hard' } })
  expect(queryByText(/drink1/)).not.toBeInTheDocument()
  expect(queryByText(/drink2/)).toBeInTheDocument()
})

test('filters correctly on base', () => {
  const { queryByText, baseFilter } = renderComponent()

  fireEvent.change(baseFilter, { target: { value: 'Gin' } })

  expect(queryByText(/drink1/)).not.toBeInTheDocument()
  expect(queryByText(/drink2/)).toBeInTheDocument()

  fireEvent.change(baseFilter, { target: { value: 'Whiskey' } })
  expect(queryByText(/drink1/)).toBeInTheDocument()
  expect(queryByText(/drink2/)).not.toBeInTheDocument()
})

test('filters correctly on flavour', () => {
  const { queryByText, flavourFilter } = renderComponent()

  fireEvent.change(flavourFilter, { target: { value: 'Ginger' } })

  expect(queryByText(/drink1/)).toBeInTheDocument()
  expect(queryByText(/drink2/)).toBeInTheDocument()

  fireEvent.change(flavourFilter, { target: { value: 'Lemonade' } })
  expect(queryByText(/drink1/)).not.toBeInTheDocument()
  expect(queryByText(/drink2/)).toBeInTheDocument()
})

test('filters correctly on search term', () => {
  const { queryByText, searchFilter } = renderComponent()

  fireEvent.change(searchFilter, { target: { value: 'wh' } })

  expect(queryByText(/drink1/)).toBeInTheDocument()
  expect(queryByText(/drink2/)).not.toBeInTheDocument()

  fireEvent.change(searchFilter, { target: { value: 'gi' } })
  expect(queryByText(/drink1/)).toBeInTheDocument()
  expect(queryByText(/drink2/)).toBeInTheDocument()
})
