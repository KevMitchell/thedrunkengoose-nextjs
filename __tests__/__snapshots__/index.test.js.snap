// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders recipe section 1`] = `
<div>
  <div
    class="jsx-1150314341"
  >
    <div
      class="jsx-3084802152 dropdown-container"
    >
      <select
        class="jsx-3084802152"
        data-testid="type-filter"
      >
        <option
          class="jsx-3084802152"
          value="all"
        >
          All types
        </option>
        <option
          class="jsx-3084802152"
          value="alcoholic"
        >
          Alcoholic
        </option>
        <option
          class="jsx-3084802152"
          value="virgin"
        >
          Virgin
        </option>
      </select>
      <span
        class="jsx-3084802152 arrow"
        tabindex="0"
      />
    </div>
    <div
      class="jsx-3084802152 dropdown-container"
    >
      <select
        class="jsx-3084802152"
        data-testid="difficulty-filter"
      >
        <option
          class="jsx-3084802152"
          value="all"
        >
          Any effort
        </option>
        <option
          class="jsx-3084802152"
          value="Easy"
        >
          Easy
        </option>
        <option
          class="jsx-3084802152"
          value="Hard"
        >
          Hard
        </option>
      </select>
      <span
        class="jsx-3084802152 arrow"
        tabindex="0"
      />
    </div>
    <div
      class="jsx-3084802152 dropdown-container"
    >
      <select
        class="jsx-3084802152"
        data-testid="base-filter"
      >
        <option
          class="jsx-3084802152"
          value="all"
        >
          All bases
        </option>
        <option
          class="jsx-3084802152"
          value="Gin"
        >
          Gin
        </option>
        <option
          class="jsx-3084802152"
          value="Whiskey"
        >
          Whiskey
        </option>
      </select>
      <span
        class="jsx-3084802152 arrow"
        tabindex="0"
      />
    </div>
    <div
      class="jsx-3084802152 dropdown-container"
    >
      <select
        class="jsx-3084802152"
        data-testid="flavour-filter"
      >
        <option
          class="jsx-3084802152"
          value="all"
        >
          All flavour
        </option>
        <option
          class="jsx-3084802152"
          value="Ginger"
        >
          Ginger
        </option>
        <option
          class="jsx-3084802152"
          value="Lemonade"
        >
          Lemonade
        </option>
        <option
          class="jsx-3084802152"
          value="Strong"
        >
          Strong
        </option>
      </select>
      <span
        class="jsx-3084802152 arrow"
        tabindex="0"
      />
    </div>
  </div>
  <div
    class="jsx-2994608941 drinkPanel "
    data-testid="drinkPanel"
  >
    <div
      class="jsx-2994608941 title"
    >
      drink1
    </div>
    <div
      class="jsx-2994608941 topPanel"
    >
      <div
        class="jsx-2994608941 overview"
      >
        <div
          class="jsx-2994608941 imageContainer"
        >
          <img
            alt="drink1"
            class="jsx-2994608941 image"
            src="./img/drinks/drink1.jpg"
          />
        </div>
        <div
          class="jsx-2994608941 overviewText"
        >
          <div
            class="jsx-2994608941 difficulty"
          >
            Easy
          </div>
          <div
            class="jsx-2994608941 base"
          >
            Whiskey
          </div>
          <div
            class="jsx-2994608941 flavour"
          >
            Strong, Ginger
          </div>
        </div>
      </div>
      <div
        class="jsx-2994608941 ingredients"
      >
        <div>
          50ml WHISKEY (BOURBON)
        </div>
        <div>
          15ml LEMON JUICE
        </div>
      </div>
      <div
        class="jsx-2994608941 methodExpander"
      >
        Method 
        ▼
      </div>
      <div
        class="jsx-2994608941 addToSaved"
      >
        <div
          class="jsx-2994608941 heart"
        />
        Add to saved
      </div>
    </div>
    <div
      class="jsx-2994608941 instructions "
      data-testid="instructions"
    >
      <div
        class="jsx-2994608941 instructionLine"
      >
        • inst1
      </div>
      <div
        class="jsx-2994608941 instructionLine"
      >
        • inst2
      </div>
    </div>
  </div>
  <div
    class="jsx-2994608941 drinkPanel "
    data-testid="drinkPanel"
  >
    <div
      class="jsx-2994608941 title"
    >
      drink2
    </div>
    <div
      class="jsx-2994608941 topPanel"
    >
      <div
        class="jsx-2994608941 overview"
      >
        <div
          class="jsx-2994608941 imageContainer"
        >
          <img
            alt="drink2"
            class="jsx-2994608941 image"
            src="./img/drinks/drink2.jpg"
          />
        </div>
        <div
          class="jsx-2994608941 overviewText"
        >
          <div
            class="jsx-2994608941 difficulty"
          >
            Hard
          </div>
          <div
            class="jsx-2994608941 base"
          >
            Gin
          </div>
          <div
            class="jsx-2994608941 flavour"
          >
            Lemonade, Ginger
          </div>
        </div>
      </div>
      <div
        class="jsx-2994608941 ingredients"
      >
        <div>
          50ml LEMONADE
        </div>
      </div>
      <div
        class="jsx-2994608941 methodExpander"
      >
        Method 
        ▼
      </div>
      <div
        class="jsx-2994608941 addToSaved"
      >
        <div
          class="jsx-2994608941 heart"
        />
        Add to saved
      </div>
    </div>
    <div
      class="jsx-2994608941 instructions "
      data-testid="instructions"
    >
      <div
        class="jsx-2994608941 instructionLine"
      >
        • inst3
      </div>
      <div
        class="jsx-2994608941 instructionLine"
      >
        • inst4
      </div>
    </div>
  </div>
</div>
`;
